### PR TITLE
fix: mark `MathOnFloatProcessor` as incomplete

### PR DESCRIPTION
**Pull request by algomaster99**
One of the ignored tests was bothering me for some time so I though I should fix it. It turned out that resolving the bug stated in #570 required some more dependent fixes and refactoring so I proceeded with them as well. Here is the summary of the list of changes in this PR.

1. `MathOnFloatProcessor` is marked as incomplete. It won't attempt to repair when the expected type of the `CtVariable` is `float` as it produces a non-compilable output.
2. Minimise the number of casts we perform. SonarSource suggests that we should cast _both_ of the operands to `double`, but that is not necessary as JVM implicitly does type coercion.
3. Use [type casting API](https://github.com/INRIA/spoon/blob/2fff4c50266f79bc06b961e3b01ca6c6e0fcc1ab/src/main/java/spoon/reflect/code/CtExpression.java#L38) in Spoon to add type cast.

### CODE REVIEW

File: sorald/src/main/java/sorald/processor/MathOnFloatProcessor.java [hunk 1]
```diff
@@ -56,12 +56,8 @@ protected void repairInternal(CtBinaryOperator element) {
...
                     binaryOperator.setType(getFactory().Type().doublePrimitiveType());
                 }
-                if (!binaryOperator.getType().equals(getFactory().Type().doublePrimitiveType())) {
-                    binaryOperator
-                            .getRightHandOperand()
-                            .setTypeCasts(List.of(getFactory().Type().doublePrimitiveType()));
-                    binaryOperator.setType(getFactory().Type().doublePrimitiveType());
-                }
+                // We do not need to cast the type of the right hand operand as it is already a
+                // double
```
Comment (algomaster99) [inline sorald/src/main/java/sorald/processor/MathOnFloatProcessor.java:60]: @monperrus not related to autocompletion, but when I deleted the above code, the inline comment wa suggested by co-pilot. 🤌🏼

**Comment by algomaster99** [comment]
Please merge this once #868 is merged into this PR.

File: sorald/src/main/java/sorald/processor/MathOnFloatProcessor.java (hunk 1 above)
Comment (khaes-kth) [reply sorald/src/main/java/sorald/processor/MathOnFloatProcessor.java:58]: Unbelievable!

**Review by khaes-kth — APPROVED** [review]

**Comment by khaes-kth** [comment]
That would be better if we could add a test to cover line 86 of this file: https://app.codecov.io/gh/SpoonLabs/sorald/compare/866/changes#D1L86

File: sorald/src/main/java/sorald/processor/MathOnFloatProcessor.java (hunk 1 above)
Comment (monperrus) [reply sorald/src/main/java/sorald/processor/MathOnFloatProcessor.java:47]: crazy

**Pull request merged**